app/lib/common/MarketClasses.js

```diff
@@ -721,12 +733,92 @@ class CallOrder {
     getCollateral() {
         if (this._collateral) return this._collateral;
         return (this._collateral = new Asset({
-            amount: this.for_sale,
-            asset_id: this.for_sale_id,
-            precision: this.assets[this.for_sale_id].precision
+            amount: this.collateral,
+            asset_id: this.collateral_id,
+            precision: this.assets[this.collateral_id].precision
         }));
     }
 
+    _getMaxCollateralToSell() {
+        /*
+        BSIP38:
+        * max_amount_to_sell = (debt * target_CR - collateral * feed_price)
+        * / (target_CR * match_price - feed_price)
+        */
+        if (this.target_collateral_ratio) {
+            let feed_price = this._getFeedPrice();
+            let match_price = this._getMatchPrice();
+            let nominator =
+                this.debt * this.target_collateral_ratio -
+                this.collateral * feed_price;
+            let denominator =
+                this.target_collateral_ratio * match_price - feed_price;
+            return nominator / denominator;
```

Comment:
I don't know if you've handled precision difference. I guess the value returned in `else` below is in Satoshi's, and what unit is returned here?

Reply:
Should be satoshis also